### DISCUSSION TITLE
OpenSumi 2022 RoadMap

### DISCUSSION THREAD

**Discussion by erha19** [opensumi/core#316] [Ideas]
在 2022 年 1 月的核心成员会议之后，我们确定了以下一些基础功能目标：

## 编辑器

- [ ] 优化 Monaco Editor 的代码补全能力。 @Ricbet

## 调试

- [ ] 优化调试控制台性能，支持调试控制台展示更多消息等。 @Ricbet
- [x] 支持新的 VS Code Javascript Debugger 插件。 @erha19 @Ricbet

## 文件系统

- [x] 支持跨窗口复制或粘贴文件，寻找跨 Web 或 Electron 窗口复制或粘贴文件的新方法。 @erha19
- [x] 将 nsfw 替换为 @parcel/watcher @bytemain @MMhunter @erha19

## 终端

- [x] 支持刷新页面后重新连接现有终端进程。 @life2015 @yantze 

## 组件

- [x] 使用@opensumi/ide-components 作为 `sumi-browser` 导出的默认组件。 @erha19
- [x] 从 `@opensumi/ide-core-common` 和 `@opensumi/ide-core-components` 中拆分一个 utils 模块。 @erha19

## 待定 

- [ ] 插件视图扩展能力采用新的新沙盒模式。 @Aaaaash
- [ ] OpenSumi 的插件市场建设。 @jinboker
~~ - [ ] 尝试兼容 Github Copilot 插件 @Aaaaash。 @hacke2 ~~
- [ ] 支持通过 Node Worker 创建插件进程获取更好的性能，例如 [microsoft/vscode#135774](https://github.com/microsoft/vscode/pull/135774) @Aaaaash 

## Electron

- [ ] 解决 URI 在 Windows 环境中的兼容问题。#232 @yantze 

## 文档

- [x] 提供更多 OpenSumi 模块拓展示例。 @hacke2 @Aaaaash @erha19
- [x] 新的官方网站`opensumi.com` 建设。 @erha19

**Comment by yantze** [comment]
感觉不是用 Features Refactor 来分类，用一些大的话题归类可能会好一些，语义化。

**Reply by erha19** [reply]
> 感觉不是用 Features Refactor 来分类，用一些大的话题归类可能会好一些，语义化。

举个例子？

**Reply by yantze** [reply]
比如文章里面：
![image](https://user-images.githubusercontent.com/2226423/149075427-605ef056-51fd-4c85-a662-faa84ad76351.png)

另外文章里面的是跨模块或者啥的就单独起一个，因为这种一般会比较多 issue，比如 Remote mode。
![image](https://user-images.githubusercontent.com/2226423/149075595-70d465fd-0c5e-4a95-b500-10d6bdbd7bd0.png)

有点类似这种： https://github.com/microsoft/vscode/issues/139607

**Reply by erha19** [reply]
可以，我晚点改一下，感觉可以按功能模块等一些细化的项分一下类

**Reply by erha19** [reply]
done

**Comment by erha19** [comment]
还有几点内容感觉可以讨论分工一下：

1. 插件研发相关工具链支持
2. 开放的 OpenSumi 插件市场
3. JAVA 研发链路的支持

几个可以思考的问题：

1. 性能还有什么改进空间？
2. 界面是否需要重新设计？
3. Welcome 及 Notebook API 是否要支持？
4. 针对 Electron M1 版本支持是否接着搞？

**Reply by Aaaaash** [reply]
5. 考虑 https://github.com/tauri-apps/tauri

**Reply by erha19** [reply]
1. 插件研发相关工具链支持 @jinboker 
2. GitLens / Git 插件升级
3. Welcome API @Aaaaash 
4. Electron M1 版本构建支持 @yantze

**Comment by Ricbet** [comment]
测试:
1. 支持新的测试交互体验（如测试资源管理器、testing decoration、output peek view 等）
2. 支持查看测试覆盖率

调试:
1. 支持 Green threads DAP
2. 改进 inline breakpoint

**Comment by erha19** [comment]
1. Electron 版本中的弹窗及确认框等使用原生能力替换

**Reply by erha19** [reply]
@yantze

**Comment by yantze** [comment]
1. 隐藏 windowId webContentsId，抽离 Electron 关于 id 相关的特性

**Reply by erha19** [reply]
清理代码

**Comment by erha19** [comment]
1. 使用最新的 ESLint 规则对全局代码进行一次改造，移除不合理的语法，如 `xxx!` 等。

**Reply by erha19** [reply]
后续暂时禁用这条，其他规则还是要保留

**Reply by bytemain** [reply]
done
